### DISCUSSION TITLE
Indexing in Oxigraph

### DISCUSSION THREAD

**Discussion by azziti** [oxigraph/oxigraph#633] [Q&A] [answered]
We have a critical use case of Oxigraph  where lookup performance is an important criteria and we need to know is indexing implemented in Oxigraph and if this is the case what techniques are used

**Comment by Tpt** [comment]
It's described on the wiki: https://github.com/oxigraph/oxigraph/wiki/Architecture

**Reply by Tpt** [reply]
If you are unhappy with the current state, help in developper time or money is very welcomed.

**Reply by azziti** [reply]
Thanks for the reply, Not really, I am just learning about RDF/Oxigraph, just asking for any advice on how to add Indexing to a triple-store, Already had a look to the arch, but not sure if it comes by default or I should implement it

**Reply by Tpt** [reply, marked as answer]
Indexing is done by the storage layer of Oxigraph. Data are stored using RocksDB. All of that is built-in, Oxigraph aim is to be as plug-and-play as Sqlite.
